### PR TITLE
Update OTel Collector to 0.50.0

### PR DESCRIPTION
**Pull request by jharley**
Address breaking changes since 0.37

### CODE REVIEW

**Comment by jharley** [comment]
Build will fail until the CI Docker changes are pushed upstream and the circle manifest is updated to use it.

File: docs/building.md [hunk 1]
```diff
@@ -5,18 +5,20 @@ Using the configuration provided by this repository requires a build of OpenTele
 * [the `hostmetrics` receiver](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/hostmetricsreceiver#readme): Lives in the contrib repository
 * [the `resourcedetection` processor](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/processor/resourcedetectionprocessor): Lives in the contrib repository
 * [the `metricstransform` processor](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/processor/metricstransformprocessor#readme): Lives in the contrib repository
+* [the `healthcheck` extension](ihttps://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/extension/healthcheckextension#readme): Lives in the contrib repository
 * [the `timestamp` processor](../timestampprocessor): Lives in this repository
```
Comment (maxedmands) [inline docs/building.md:9]: Is this necessary anymore? As we're doing this work in shepherd now.

**Review by maxedmands — APPROVED** [review]
Woah, lots has changed, thanks for updating the repo!

File: docs/building.md (hunk 1 above)
Comment (jharley) [reply docs/building.md:9]: I'm torn.. I know some teams are using it with `round_to_nearest` set to 10s, 15s, or 30s and I think we're doing 1s 'bucketing' on ingest now?

Comment (maxedmands) [reply docs/building.md:9]: That's correct. If folks want to up the number, this is still valuable! Wasn't aware that was an active use-case :)

**Review by vreynolds — APPROVED** [review]

**Pull request merged**